padl/transforms.py

```diff
@@ -324,28 +325,41 @@ def __invert__(self) -> "Map":
         """
         return Map(self)
 
-    def pd_pre_save(self, path: Path, i: int):
+    def pd_pre_save(self, path: Path, i: int, options: Optional[dict] = None):
         """Method that is called on each transform before saving.
 
         This normally does nothing. Override to implement custom serialization.
 
         :param path: The save-folder path.
         :param i: Unique transform index, can be used to construct filenames.
         """
+        opt = None
+        if options is not None:
+            opt = self._pd_process_options(options)
+        # pd_pre_save requires default behaviour on receiving None
         try:
+            if 'options' in inspect.signature(self.pre_save).parameters:
```

Comment:
Didn't we want to replace `try..catch AttributeError` with `if hasattr(self, 'pre_save')` to prevent catching of attribute errors inside `pre_save`?